### ISSUE TITLE
MAE Spatiotemporal Learners (Garbage Results)

### ISSUE THREAD

**Issue by innat** [facebookresearch/SlowFast#668]
**Background**

I was following the interesting work done by facebookresearch on [Masked Autoencoders As Spatiotemporal Learners](https://github.com/facebookresearch/mae_st/tree/main) which later added to [SlowFast/mae_st](https://github.com/facebookresearch/SlowFast/blob/main/projects/mae/README.md). With the provided pretrained weight, I tried to call inference on series of video frame (a smple from kinetric400), however, the results with the pretrained model gives totally nonsene prediction. 

![image](https://github.com/facebookresearch/SlowFast/assets/17668390/cd0a4331-bf85-42c5-b8d9-e2f06f5996b0)

**Reproducible Code**

Please find the gist here. [COLAB](https://colab.research.google.com/drive/1L96PXqotW5aPp4KQKrdhjadWxXBU0w3h?usp=sharing)

**Info**

- I have ported the model definition from `mae_st` repo as it was compratively cleaner and to the point.
- The pre-trained weight links of both `mae_st` and `slowfast_mae_st` repos are tried, but gives nonsene prediction.
- The above colab is **ready-to-go**; just run all cell.

**Comment by innat** [comment]
Hi @r-barnes
I've just noticed some update on [mae_st](https://github.com/facebookresearch/mae_st) repo. By any chance, could you please inform if the [provided weights](https://github.com/facebookresearch/mae_st#fine-tuning-with-pre-trained-checkpoints) are in good health?

**Comment by alpargun** [comment]
Looking at the collab, you have a long list of missing keys, i.e., not restored layers, when you load the checkpoint and state dictionary:

<img width="1588" alt="image" src="https://github.com/facebookresearch/SlowFast/assets/52760747/c7e0fac9-7e38-4694-b5a9-53c8da288f9d">

So, you are running inference with random weights.

The vision models in SlowFast, timm, MAE_ST, torchhub, etc., do not necessarily (very rarely) have perfectly matching implementations. Mostly the layer names, i.e., state dictionary keys, will differ in name. So, I would suggest you to run inference using directly the SlowFast framework, including the corresponding config file, and class implementations.

If you have troubles restoring the model, I can provide more information or code snippets.

**Comment by innat** [comment]
Hello @alpargun , I stopped working on it long ago and forgot to close this ticket. I emailed to one of the authors but no response so far. However, if you are able to fix the colab issue, and fix cost is minimal, please go ahead. I highly appreciate your help. Thank you.
